Automattic/happychat-client
# Receive and display optimistic messages

In 1798-gh-Automattic/happychat we change message routing so that an optimistic version is sent back to the user in `message.optimistic`, followed by the canonical `message` event after the message had persisted. This PR listens for those optimistic events and inserts them into the timeline, being replaced by the canonical version when it comes in.

There's no visual change for users, but I added an `.is-optimistic` class since it's likely we'll want to use that soon (e.g. when we add Image Uploading, we'll need to display "placeholder" images for optimistic messages, to be replaced with the actual image later).

### How to test

When used with the current `happychat` service nothing should change for the user experience. Messages should be received immediately from both yourself and operators. So this PR can deploy in isolation from the socket server update.

When used with 1798-gh-Automattic/happychat the experience should be the same when you send messages — you'll see them appear right back in your timeline. However there will be a _slight_ delay between operators sending messages, and those messages being shown in the customer timeline — because the customer does not receive an optimistic version of operator messages.
